fgrpc/grpcrunner.go

```diff
@@ -316,14 +316,26 @@ func grpcDestination(dest string) (parsedDest string) {
 func extractDialOptions(in metadata.MD) (out []grpc.DialOption) {
```

Comment:
seems maybe this should just return 2 things, the dial options and the filtered MD? (avoiding the separate sanitize() or at least rename sanitize for something clearer like filterMetaData() but it seems odd the 2 can't be combined)

Reply:
If I understand this correctly, then in terms of function naming, the "Dial" used for the connection should not contain side effects

Reply:
ic the issue is the one time setup is here but somehow the grpc meta data keeps getting set in the context each time, it's not very efficient to keep "sanitizing" it though even if we keep assigning it

Reply:
@ktalg please look at https://github.com/fortio/fortio/pull/701/commits/416e9d5e0a3781a95efa31e1aede251ff6bdd0cd for what I meant

(tried to push to your branch but failed so it's in 701)

Reply:
> ic the issue is the one time setup is here but somehow the grpc meta data keeps getting set in the context each time, it's not very efficient to keep "sanitizing" it though even if we keep assigning it

I'm aware of this, and it seems like a much bigger change would be needed to handle it well, and I have to be honest and say I can't assess whether such a change would be worth it 🥲 It involves a lot of design considerations...

Reply:
well I did the change, can you review it in #701

Reply:
I will try again to push it to your fork in a bit

Reply:
@ktalg it worked this time, PTAL at https://github.com/fortio/fortio/pull/697/commits/d906484c90afbcc25ed1d17aa2010660916c5402
and if you think it makes sense we can merge this and put it in 1.40.1

Reply:
> @ktalg it worked this time, PTAL at [d906484](https://github.com/fortio/fortio/commit/d906484c90afbcc25ed1d17aa2010660916c5402) and if you think it makes sense we can merge this and put it in 1.40.1

it makes sense, thanks a lot!